Packages/StatusKit/Sources/StatusKit/Row/Subviews/StatusRowCardView.swift

```diff
@@ -91,28 +84,48 @@ public struct StatusRowCardView: View {
     .buttonStyle(.plain)
   }
 
-  @ViewBuilder
-  private func defaultLinkPreview(_ title: String, _ url: URL) -> some View {
-    if let imageURL = card.image, !isInCaptureMode {
-      LazyResizableImage(url: imageURL) { state, proxy in
-        let width = imageWidthFor(proxy: proxy)
-        if let image = state.image {
-          image
-            .resizable()
-            .aspectRatio(contentMode: .fill)
-            .frame(height: imageHeight)
-            .frame(maxWidth: width)
-            .clipped()
-        } else if state.isLoading {
-          Rectangle()
-            .fill(Color.gray)
-            .frame(height: imageHeight)
+  struct DefaultCardImage: View {
+    let url: URL
+    let maxWidth: CGFloat?
+    let defaultImageHeight: CGFloat
+    let theme: Theme
+
+    @State private var width: CGFloat? = nil
+    @State private var height: CGFloat? = nil
+
+    var body: some View {
+      LazyResizableImage(url: url) { state, proxy in
+        let proxyWidth = imageWidthFor(proxy: proxy)
+        if let image = state.image,
+           let cgImage = state.imageContainer?.image.cgImage
+        {
+          image.resizable().scaledToFit()
+            .onAppear {
+              self.width = proxyWidth
```

Comment:
The thing I don't like about that is that it's costly. It'll create stutters as as the image appears. I wish we could solve that differently, without relying to onAppear.

Reply:
I don't think I can let that in. I agree it's better and how it's supposed to look but with SwiftUI List it creates performance issues. As stuff get resized while scrolling.

Reply:
Actually the API state what the should have `width` and `height` in `Card`. Maybe you could add this and we could calculate its size statically. We could even show the square preview if it's a square.